### PR TITLE
STN-737  Photo album slider

### PR DESCRIPTION
**Pull request by rkelbel**
# READY FOR REVIEW

## Description
I want to add images to my site so that I can showcase photos through a slider display. 

JIRA Card: https://sparkbox.atlassian.net/browse/STN-737

## Summary of Work
- The work to implement the photo gallery Slider has already been implemented by Mike. You can find an example of it on staging.
- Slick:
    - A new Slick Slideshow option set has been created for this Slideshow feature. (It is a clone of our former option set we used on Carousel and Gradient Hero Slider, but with the Dot selectors turned off as they are not needed in this slider.)
- The Slider headline, description, and link should be styled to match our current styles.
- Controls: West confirmed using the > arrow style and the semi-circle overlay for the slider design (circled below).
_The controls should feature caret icons positioned to the right and the left of the carousel container. The caret icons are placed within a semi-circle with white background at 50% opacity. The color of the caret icons on Colorful are primary and the color of the caret icons on Traditional are global black._
- The image aspect ratio should be set through the configuration Manage Display settings with an appropriate pixel ratio of 4:3
    - New Image syles have been created to keep this ratio within responsive media queries.
- The existing caption/credits styling (view mode) should be used and set as the default through the configuration Manage Display
    - Created a new View Mode that also uses Caption Credit that uses Responsive Image View Mode to keep the proper ratio size.

## Need Review By (Date)
6/10

## Urgency
medium

## Steps to Test
1. npm run test
2. `lando drush @sparkbox_sandbox.local cim --partial` (I noticed a new module was added in the release, you may need to do a composer install and sync prior to running this command on this branch.)
3. Turn on the Paragraph Style "Photo Album" on Flexible Page Content Type.
4. Create a Flexible Page with a Photo Album, which needs to be turned on to the Slideshow option vs. Grid.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  .su-gallery-button a {
+    @include hb-button;
+    margin-top: hb-calculate-rems(32px);
+  }
+
+  // Photo Gallery Slideshow
+  .hs-gallery-slideshow {
+    @include slick-container;
+    @include slick-arrows;
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss:20]: I'm so excited these mixins worked as I had hoped!!! 😄

File: docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// This styling is for the Photo Gallery Grid and Photo Gallery Slideshow
+.ptype-stanford-gallery {
+  .su-gallery-images article,
+  .slick__slide article {
+    position: relative;
+  }
+
+  .su-gallery-description {
+    margin-bottom: hb-calculate-rems(32px);
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss]: Can we use `margin-bottom: hb-spacing-width('md');` here? That will allow Traditional theme to have the tighter editorial spacing in comparison to the Colorful theme.

File: docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  .su-gallery-button a {
+    @include hb-button;
+    margin-top: hb-calculate-rems(32px);
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss]: Same here. Can we update to `margin-top: hb-spacing-width('md');`?

File: docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss (hunk 1 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss:20]: After testing on mobile devices I think we should keep the arrows (caret icons) to the left and right of the gallery image at all screen sizes. Here's a screenshot of a tablet view:

keep icons to right and left at all screen sizes

And a screenshot of a mobile view at 320px:

mobile 320px

**Review by heatherdesigns — COMMENTED** [review]
## Fantastic work!!! :tada: :tada: :tada:

cute high five

I left a couple recommendations for styling tweaks within the code then this is ready to roll!

If it's possible, when there is only one image added to a photo gallery can we have it span 100% of the width of the main content area? I'm not sure what the chance is someone might add a single image, but just in case. :smile:

single photo in album

**Comment by rkelbel** [comment]
> If it's possible, when there is only one image added to a photo gallery can we have it span 100% of the width of the main content area? I'm not sure what the chance is someone might add a single image, but just in case. 😄

This looks like this is set to the Grid part of the photo gallery, when I have one slide set to the Slider, the one image goes full width and doesn't have the controls as expected.

image

File: docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss (hunk 1 above)
Comment (rkelbel) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss:20]: I adjusted the carousel arrows for mobile size and added a media query for the size at the small size for the small tablet size before it regulates to the normal desktop controls.

Based on our conversation at standup that H&S wanted to keep the buttons on one side for consistency. _(Noting that just for documenting it.)_

File: docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    .slick__arrow button {
+      &.slick-prev {
+        bottom: hb-calculate-rems(98px);
+
+        @include grid-media-min('sm') {
+          bottom: hb-calculate-rems(198px);
+        }
+      }
+      &.slick-next {
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss:31]: Linting error:

line please

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.general.scss [hunk 5]
```diff
@@ -38,7 +38,8 @@
 .ptype-hs-row,
 .ptype-hs-timeline,
 .ptype-hs-view,
-.ptype-hs-collection {
+.ptype-hs-collection,
+.ptype-stanford-gallery {
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.general.scss:42]: Good call! Sometimes I forget to add components here. 🥇

File: docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss (hunk 1 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss:20]: Mobile looks great!!!

**Review by heatherdesigns — COMMENTED** [review]
One small change then this is ready to roll!

parrot

File: docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss (hunk 4 above)
Comment (rkelbel) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.photo-album-slider.scss:31]: Yep, good catch, sorry I missed that.
All fixed up now!

**Review by heatherdesigns — APPROVED** [review]

**Pull request merged**